x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRestoreSourceService.java

```diff
@@ -133,6 +133,17 @@ public synchronized void closeSession(String sessionUUID) {
         IOUtils.closeWhileHandlingException(restore);
     }
 
+    // TODO: The Engine.IndexCommitRef might be closed by a different thread while it is in use. We need to
+    //  look into the implications of this.
```

Comment:
Similar to the `RecoveryTarget` object, we could use some ref-counting here (see `AbstractRefCounted`)